### ISSUE TITLE
Product resource getAttributeRawValue function does not return anything if there is no default value

### ISSUE THREAD

**Issue by VincentMarmiesse** [magento/magento2#16382]
### Preconditions

1. Magento 2.2.4
2. PHP 7.1.17

### Steps to reproduce
1. Create a storeview EN
2. Create a storeview FR
3. Create a dropdown attribute with some values
4. Change this attribute value on EN view
5. Use `getAttributeRawValue()` to retrieve it, ie 
```
$product->getResource()->getAttributeRawValue($product->getId(), 'attribute_code', $this->_storeManager->getStore()->getId());
```

### Expected result

1. Value is returned

### Actual result

1. No value is returned because there is no value in default store (store 0).

**Comment by magento-engcom-team** [comment]
Hi @VincentMarmiesse. Thank you for your report. 
To help us process this issue please make sure that you provided the following information:
- [x] Summary of the issue
- [x] Information on your environment
- [x] Steps to reproduce
- [x] Expected and actual results

Please make sure that the issue is reproducible on the vanilla Magento instance following **Steps to reproduce**. To deploy vanilla Magento instance on our environment, please, add a comment to the issue: 

`@magento-engcom-team give me {$VERSION} instance` 

where `{$VERSION}` is [version tags](https://github.com/magento/magento2/tags) (starting from 2.2.0+) or develop branches (2.2-develop +). 
For more details, please, review the [Magento Contributor Assistant](https://github.com/magento/magento2/wiki/Magento-Contributor-Assistant) documentation.

@VincentMarmiesse do you confirm that you was able to reproduce the issue on vanilla Magento instance following steps to reproduce?
- [ ] yes
- [ ] no

**Comment by VincentMarmiesse** [comment]
@magento-engcom-team give me 2.2.4 instance

**Comment by magento-engcom-team** [comment]
Hi @VincentMarmiesse. Thank you for your request. I'm working on Magento 2.2.4 instance for you

**Comment by magento-engcom-team** [comment]
Hi @VincentMarmiesse, here is your [Magento instance](https://i-16382-2-2-4.engcom.dev.magento.com).
Admin access: [https://i-16382-2-2-4.engcom.dev.magento.com/admin](https://i-16382-2-2-4.engcom.dev.magento.com/admin)
Login: `admin` Password: `123123q`
Instance will be terminated in up to 3 hours.

**Comment by VincentMarmiesse** [comment]
I can't confirm with this instance because I have to write some code and call the function `getAttributeRawValue`.
Can I do it with your instance?

**Comment by orlangur** [comment]
@VincentMarmiesse is it a duplicate of https://github.com/magento/magento2/issues/15205 maybe? Please check if corresponding PR worked for you.

**Comment by VincentMarmiesse** [comment]
@orlangur no it's not. It's not related to the session or translation, the SQL request done by `getAttributeRawValue` is assuming there is a value for the store 0, but it's not always the case.

**Comment by engcom-backlog-nickolas** [comment]
Hello @VincentMarmiesse, thank you for your report.
We've acknowledged the issue and added to our backlog.

**Comment by magento-engcom-team** [comment]
Hi @shikhamis11. Thank you for working on this issue. 
Looks like this issue is already verified and confirmed. But if your want to validate it one more time, please, go though the following instruction:
- [ ] 1. Add/Edit `Component: XXXXX` label(s) to the ticket, indicating the components it may be related to.

- [ ] 2. Verify that the issue is reproducible on `2.3-develop` branch<details><summary>Details</summary>- Add the comment `@magento-engcom-team give me 2.3-develop instance` to deploy test instance on Magento infrastructure. <br/> - If the issue is reproducible on `2.3-develop` branch, please, add the label `Reproduced on 2.3.x`.<br/>- If the issue is not reproducible, add your comment that issue is not reproducible and close the issue and **_stop verification process here_**!
</details>

- [ ] 3. Verify that the issue is reproducible on `2.2-develop` branch. <details><summary>Details</summary>- Add the comment `@magento-engcom-team give me 2.2-develop instance` to deploy test instance on Magento infrastructure. <br/>- If the issue is reproducible on `2.2-develop` branch, please add the label `Reproduced on 2.2.x`</details>

- [ ] 4. If the issue is not relevant or is not reproducible any more, feel free to close it.

**Comment by shikhamis11** [comment]
#MM19IN

**Comment by dmzaiets** [comment]
@magento-engcom-team give me 2.2.4 instance

**Comment by magento-engcom-team** [comment]
Hi @funnyboy1190. Thank you for your request. I'm working on Magento 2.2.4 instance for you

**Comment by magento-engcom-team** [comment]
Hi @funnyboy1190, here is your [Magento instance](https://i-16382-2-2-4.instances.magento-community.engineering).
Admin access: [https://i-16382-2-2-4.instances.magento-community.engineering/admin](https://i-16382-2-2-4.instances.magento-community.engineering/admin)
Login: `admin` Password: `123123q`
Instance will be terminated in up to 3 hours.

**Comment by semajeg** [comment]
#mmuk2019

**Comment by magento-engcom-team** [comment]
Hi @VincentMarmiesse. Thank you for your report. 
The issue has been fixed in magento/magento2#23369 by @semajeg in 2.3-develop branch
Related commit(s):
 - [01f1f157dc2f17a0bbefcfe9ba717c96d788fb53](https://github.com/magento/magento2/commit/01f1f157dc2f17a0bbefcfe9ba717c96d788fb53)
 - [64184b3bf28b2a88ac691ca651d219ed74cf94ef](https://github.com/magento/magento2/commit/64184b3bf28b2a88ac691ca651d219ed74cf94ef)
 - [35c5185bdae84cae7ac5008098ae0d2d893cf05b](https://github.com/magento/magento2/commit/35c5185bdae84cae7ac5008098ae0d2d893cf05b)
 - [9ae33c855dd68bbdfcfe4148facaaa9c0c4b30a1](https://github.com/magento/magento2/commit/9ae33c855dd68bbdfcfe4148facaaa9c0c4b30a1)
 - [8c4bfa62528235b651a06e9a24f4a8a0b3a5f7f9](https://github.com/magento/magento2/commit/8c4bfa62528235b651a06e9a24f4a8a0b3a5f7f9)
 - [7b0e528a316af06dac9d6327db25ed53065a0893](https://github.com/magento/magento2/commit/7b0e528a316af06dac9d6327db25ed53065a0893)
 - [c34f83cb30a5960a62dbeff562fdbf7215d616a0](https://github.com/magento/magento2/commit/c34f83cb30a5960a62dbeff562fdbf7215d616a0)
 - [4d380bf488e027047833d270bb7d816d78d0f43f](https://github.com/magento/magento2/commit/4d380bf488e027047833d270bb7d816d78d0f43f)
 - [540f03e33772b366205c3862cc9c44a125f110ea](https://github.com/magento/magento2/commit/540f03e33772b366205c3862cc9c44a125f110ea)

The fix will be available with the upcoming 2.3.4 release.
